vault/identity_store_oidc_provider.go

```diff
@@ -1238,15 +1243,28 @@ func (i *IdentityStore) pathOIDCReadProvider(ctx context.Context, req *logical.R
 		return nil, nil
 	}
 
+	issuer := i.getFullIssuerURI(ns.Path, provider.Issuer, name)
```

Comment:
Nit: I think the result of `getFullIssuerURI()` would already be set to `provider.effectiveIssuer` from the call on L1238? If so, we could remove this call and use `provider.effectiveIssuer` in the response data.

If that works out, we could delete the `getFullIssuerURI()` function, since it'd only be called from one place. If you think there is some utility in keeping it around, I'm also okay with that 👍